### ISSUE TITLE
mini.tabline to work with files opened in tabs

### ISSUE THREAD

**Issue by shubham-cpp** [echasnovski/mini.nvim#19]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### Module(s)

mini.tabline

### Currently mini.tabline only supports buffers and stops working if we created tabs

![output](https://user-images.githubusercontent.com/31388020/142835313-44f8083d-d9d2-457c-9389-6ce161bff708.gif)
.

### Maybe provide a way to add custom modules(functions) to tab like we can add custom modules to statusline?

**Comment by echasnovski** [comment]
Yeah, fallback to default tabline in case of multiple tabs is an initial design. Mainly because of two reasons: simplify module's logic and lack of certainty what should be displayed with several tabs. And also I don't really use tabs, so didn't have motivation to work on it.

What would you expect to be shown in tabline when there are multiple tabs?

**Comment by shubham-cpp** [comment]
1. Same nothing much changed. Show icon and only filename
2. In my current lualine setup I have far right of tabline to show complete path of file. For this I have created a custom function.

![image](https://user-images.githubusercontent.com/31388020/142837604-43bcfadb-7b5a-4a37-ba13-eb0e21c19ce0.png)

I find tabs very useful, whenever working on a project I'm like Tab1 is for javascript; Tab2 is for html; Tab3 for css

**Comment by echasnovski** [comment]
Could you, please, elaborate a little about 1? Show name of current file or all files in current tab or all listed files (same as no tabs)? Or something else?

Anyway, thanks for the feedback. I'll think about it, but I don't see it being on top of my list for quite some time.

**Comment by shubham-cpp** [comment]
> Could you, please, elaborate a little about 1? Show name of current file or all files in current tab or all listed files (same as no tabs)? Or something else?

Just like normal tabs, the tabline component works fine on single tab, just want it work the same on mutiple tabs

> Anyway, thanks for the feedback. I'll think about it, but I don't see it being on top of my list for quite some time.

Yeah sure right now I'll just use regular tabline, since I've already switched to mini.statusline.

**Comment by zim0369** [comment]
@echasnovski I think you should also make little UI changes. I've experienced that as soon as you make changes to a file then the currently highlighted tab(buffer) becomes dim & then looks similar to color of other tabs while something like bufferline plugin shows a green filled circle.

I've seen this behavior with multiple themes.

**Comment by echasnovski** [comment]
> @echasnovski I think you should also make little UI changes. I've experienced that as soon as you make changes to a file then the currently highlighted tab(buffer) becomes dim & then looks similar to color of other tabs while something like bufferline plugin shows a green filled circle.
> 
> I've seen this behavior with multiple themes.

The reason you see this is because by default all highlight groups are linked to Neovim's builtin highlight groups in order to ensure that they are somehow aligned with your color scheme. in 'mini.tabline' they are linked [like this](https://github.com/echasnovski/mini.nvim/blob/6cb7cdb1cd7f111784d5d971fa41a655e11df4b3/lua/mini/tabline.lua#L102-L110), which seem like a reasonable choice to make a cohesive set of highlights. In ideal world, all colorschemes out there would overwrite these highlight groups to make them more appropriate to its visual style. Unfortunately, this is not the case.

At the moment, you have the following options:
- Manually customize highlight groups you don't like.
- Use 'mini.base16' with palette of your choice. It supports all 'mini.nvim' highlight groups.
- Propose changes to a colorscheme of your choice through a PR or wait for me to do it for most popular colorschemes out there (it is next on my TODO list).

**Comment by echasnovski** [comment]
After a third look at it, I decided to introduce breaking change and show buffer list also when multiple tabpages are open. I quite rarely use them, but it indeed feels inconsistent to fall back to default tabline (hence a breaking change and not an option).

To show that there are multiple tabs open, there is a "tabpage section": displays number of current tab and total number of tabs. Its position can be configured via `tabpage_section` option.

All in all, thanks @shubham-cpp for useful suggestion.
